ddtrace/context.py

```diff
@@ -123,11 +131,57 @@ def sampling_priority(self, value):
     @property
     def _traceparent(self):
         # type: () -> str
-        if self.trace_id is None or self.span_id is None:
-            return ""
+        tp = self._meta.get(_TRACEPARENT_KEY)
+        if self.span_id is None or self.trace_id is None:
+            # if we only have a traceparent then we'll forward it
+            # if we don't have a span id or trace id value we can't build a valid traceparent
+            return tp or ""
+
+        # determine the trace_id value
+        if tp:
+            # grab the original traceparent trace id, not the converted value
+            trace_id = tp.split("-")[1]
+        else:
+            trace_id = "{:032x}".format(self.trace_id)
 
         sampled = 1 if self.sampling_priority and self.sampling_priority > 0 else 0
-        return "00-{:032x}-{:016x}-{:02x}".format(self.trace_id, self.span_id, sampled)
+        return "00-{}-{:016x}-{:02x}".format(trace_id, self.span_id, sampled)
+
+    @property
+    def _tracestate(self):
+        # type: () -> str
+        dd = []
+        if self.sampling_priority:
+            dd.append("{}:{}".format(_W3C_TRACESTATE_SAMPLING_PRIORITY_KEY, self.sampling_priority))
+        if self.dd_origin:
+            # the origin value has specific values that are allowed.
+            dd.append("{}:{}".format(_W3C_TRACESTATE_ORIGIN_KEY, re.sub(r",|;|=|[^\x20-\x7E]+", "_", self.dd_origin)))
+        sampling_decision = self._meta.get(SAMPLING_DECISION_TRACE_TAG_KEY)
+        if sampling_decision:
+            dd.append("t.dm:{}".format(re.sub(_W3C_TRACESTATE_INVALID_CHARS_REGEX, "_", sampling_decision)))
+        # since this can change, we need to grab the value off the current span
+        usr_id_key = self._meta.get(USER_ID_KEY)
+        if usr_id_key:
+            dd.append("t.usr.id:{}".format(re.sub(_W3C_TRACESTATE_INVALID_CHARS_REGEX, "_", usr_id_key)))
```

Comment:
nit: we can probably move this to a utils function too and name it `_w3c_format_known_propagated_tags`?